libpod/container_internal.go

```diff
@@ -313,16 +313,111 @@ func (c *Container) save() error {
 	return nil
 }
 
+// Initialize a container, creating it in the runtime
+func (c *Container) init() error {
+	if err := c.makeBindMounts(); err != nil {
+		return err
+	}
+
+	// Generate the OCI spec
+	spec, err := c.generateSpec()
+	if err != nil {
+		return err
+	}
+
+	// Save the OCI spec to disk
+	if err := c.saveSpec(spec); err != nil {
+		return err
+	}
+
+	// With the spec complete, do an OCI create
+	if err := c.runtime.ociRuntime.createContainer(c, c.config.CgroupParent); err != nil {
+		return err
+	}
+
+	logrus.Debugf("Created container %s in OCI runtime", c.ID())
+
+	c.state.State = ContainerStateCreated
+
+	return c.save()
+}
+
+// Initialize (if necessary) and start a container
+// Performs all necessary steps to start a container that is not running
+// Does not lock or check validity
+func (c *Container) initAndStart() (err error) {
+	// If we are ContainerStateUnknown, throw an error
+	if c.state.State == ContainerStateUnknown {
+		return errors.Wrapf(ErrCtrStateInvalid, "container %s is in an unknown state", c.ID())
+	}
+
+	// If we are running, do nothing
+	if c.state.State == ContainerStateRunning {
+		return nil
+	}
+	// If we are paused, throw an error
+	if c.state.State == ContainerStatePaused {
+		return errors.Wrapf(ErrCtrStateInvalid, "cannot start paused container %s", c.ID())
```

Comment:
nit of a nit.  "paused container" to "a paused container"

Reply:
I disagree, since this would read
cannot start paused container ABC124
which is better then
cannot start a paused container ABC124

Reply:
DOH!  Yep, you're right, I completely had a dyslexic moment and missed the ctrid.  Ignore this comment.